src/client/components/pages/parts/privs-edit-modal.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2023 Shivam Awasthi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import {PrivilegeTypes, getPrivilegeShieldIcon, getPrivilegeTitleFromBit} from '../../../../common/helpers/privileges-utils';
+import {faPencilAlt, faTimes} from '@fortawesome/free-solid-svg-icons';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PrivilegeBadges from './privilege-badges';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+
+const {Button, Form, Modal} = bootstrap;
+
+class PrivsEditModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			privs: props.targetUser.privs,
+			submittable: false
+		};
+
+		this.handleBitChange = this.handleBitChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+	async handleSubmit() {
+		const {privs} = this.state;
+		const oldPrivs = this.props.targetUser.privs;
+		if (privs === oldPrivs) {
+			return;
+		}
+
+		const data = {
+			adminId: this.props.adminId,
+			newPrivs: privs,
+			oldPrivs,
+			targetUserId: this.props.targetUser.id
+		};
+
+		try {
+			const response = await fetch('/editor/privs/edit/handler', {
+				body: JSON.stringify(data),
+				headers: {
+					'Content-Type': 'application/json; charset=utf-8'
+				},
+				method: 'POST'
+			});
+
+			if (!response.ok) {
+				const {error} = await response.json();
+				throw new Error(error ?? response.statusText);
+			}
+			this.props.updateResultsOnPrivsChange();
+			this.props.handleCloseModal();
+		}
+		catch (err) {
+			throw new Error(err);
+		}
+	}
+
+	/* eslint-disable no-bitwise */
```

Comment:
You know, there's rarely a good use for disabling linting rules, but if I ever need a good example this will be the perfect one ! :D